### DISCUSSION TITLE
User defined Javascript functions

### DISCUSSION THREAD

**Discussion by MrGonzo1024** [Budibase/budibase#5138] [Help]
Hi,

I'm enjoying Budibase so far. It's a great project and great support.

I have a question on where to put my own javascript functions for later use in Repeaters and other containers throughout my application. I apologize if this was asked and answered somewhere. I can't seem to find an answer in the forum, but maybe I'm not looking in the right place. 

For example, I have this function that I use to format currency values:

function formatMoney(n) {
    return "$ " + (Math.round(n * 100) / 100).toLocaleString();
}

It works well when I put it in the Javascript section within a component. But it only works for the Javascript within that one component, it's not callable for any other component, unless I cut and paste it. Which would not be a very good practice. Is there a place I can put my own custom utility functions such as this one where I can call it from within any component?

Thanks!

**Comment by shogunpurple** [comment]
Hi @MrGonzo1024 - thanks very much for the kind words.

This does not exist today, however this is a great suggestion and something that has been raised before, although not under an obvious name. Thank you very much for searching through the discussions and issues first before raising this one 🙏 

https://github.com/Budibase/budibase/issues/3571

This is absolutely something that we are going to do, and it's on the roadmap for this year (Q3). This feature will allow users to have a full gallery of snippets that they can save, and reuse across the budibase platform.

**Reply by MrGonzo1024** [reply]
Thanks @shogunpurple ! Looking forward to that update.
